instance/github.py

```diff
@@ -149,26 +140,48 @@ class PR:
     """
     Representation of a GitHub Pull Request
     """
-    # pylint: disable=too-many-arguments
-    def __init__(self, number, source_fork_name, target_fork_name, branch_name, title, username, body=''):
+
+    def __init__(self, target_fork_name, number):
         self.number = number
-        self.fork_name = source_fork_name
         self.repo_name = target_fork_name
-        self.branch_name = branch_name
-        self.title = title
-        self.username = username
-        self.body = body
+        self.update_data()
+
+    def update_data(self):
+        self._data = get_pr_by_number(target_fork_name, number)
+        self._extra_settings = get_settings_from_pr_body(self.body)
+        self._extra_settings_dict = yaml.load(self.extra_settings) or {}
```

Comment:
Here, it makes sense to cache `self._data`, since this is retrieved from an external service.

For `self._extra_settings` and `self._extra_settings_dict`, the extra cache layers seem like premature optimization, if this is just for performance concerns. To keep with our previous conversation about the data structure here, I'd rather duplicate data as little as possible, ie only when it proves to be necessary; I don't think we are getting any performance constraints from generating these as properties, no?